docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+- *Space Manager*: This module manages the lifecycle of spaces.
 
-1. OCM for inventory and transport
-2. A central controller-manager that provides the added functionalities
+- *OCM Cluster Manager*: This module syncs objects from the ITS to the Workload Execution 
+Clusters (WECs). In the ITS, each mailbox namespace is associated with one WEC. Objects 
+that are put in a mailbox namespace are delivered to the matching WEC.
 
-We plan to make the transport pluggable in the near future.
+- *Status Add-On Controller*: This module installs the OCM status add-on agent 
+on all WECs and sets the RBAC permissions for it using the OCM add-on framework.
 
+- *OCM Agent*: This module registers the WEC to the OCM Hub, watches for 
+ManifestWorks and unwraps and syncs the objects into the WEC.
+
+- *OCM Status Add-On Agent*: This module watches *AppliedManifestWorks* 
+to find objects that are synced by the OCM agent, gets their status 
+and updates *workstatuses* objects in the ITS namespace associated with the WEC.
+
+<figure>
+  <img src="assets/main-modules.png"  alt="Main Modules">
+  <figcaption align="center">Figure 2 - Main Modules </figcaption>
+</figure>
+
+## KubeStellar Controller Manager
+
+This module manages the placement and status controllers. The placement controller watches 
+placement and workload objects on the Workload Definition Space 
+(WDS) and wraps workload objects into a manifest for delivery through the 
+Inventory and Transport Space (ITS). The status controller watches for `workstatuses` objects on the 
+ITS and updates the status of objects in the WDS when singleton status is requested 
+in the placement for those objects. There is one instance of a KubeStellar Controller 
+Manager for each WDS. Currently this controller-manager runs in the KubeFlex hosting cluster.
+More details on the internals of this module are provided in [KubeStellar Controllers Architecture](#kubestellar-controllers-architecture)
+
+## Space Manager
+
+The Space Manager handles the lifecycle of spaces. 
+KubeStellar uses the [KubeFlex project](https://github.com/kubestellar/kubeflex)
+for space management. In KubeFlex, a space is named a `ControlPlane`, and we will use 
+both terms in this document. KubeSteller currently prereqs KubeFlex to 
+provide one or more spaces. We plan to make this optional in the near future.
+
+KubeFlex is a flexible framework that supports various kinds of control planes, such as:
+
+- *k8s*: a basic Kubernetes API Server with a subset of kube controllers. 
+This control plane does not run workloads (e.g., pods), and 
+it uses about 350 MB of memory per instance with a shared Postgres Database Backend.
+- *vcluster*: a virtual cluster that runs on the hosting cluster, 
+based on the  [vCluster Project](https://www.vcluster.com). This type control plane can run pods using worker nodes of the hosting cluster.
+- *host*: the KubeFlex hosting cluster, which is exposed as a control plane.
```

Comment:
Do we need to say something about the denaturing capabilities?   We knows that k8s API server is a space , however a space is not necessarily a cluster .. So there is a difference in what you can do with each such "space"

Reply:
added some clarification here.